packages/service/src/service/main.ts

```diff
@@ -218,6 +220,37 @@ export function createTSLanguageService(initOptions: TSLanguageServiceOptions) {
 
       return results.length > 0 ? results : null;
     }),
+    diagnostics: wrapRequestHandler(async (params: lsp.DocumentDiagnosticParams): Promise<lsp.DocumentDiagnosticReport> => {
+      const doc = getOpenedDoc(params.textDocument.uri);
+      delegate.logMessage(lsp.MessageType.Log, `diagnostics: ${doc.uri.fsPath}`);
+      // Execute and wait for geterr to finish. While it executes, tsserver sends the diagnostic events to the client.
+      await shims.workspaceService.waitForGetErr();
+      // There is a 50ms debounce between receiving diagnostics and reporting them to vtsls.
+      // See packages/service/vscode/extensions/typescript-language-features/src/languageFeatures/diagnostics.ts#L392 
+      await new Promise(resolve => setTimeout(resolve, 100))
```

Comment:
do we need to `waitForGetErr` again after this? (or rather, if so, you want to put this into the `while` there or something)